### PR TITLE
RA-1714:Primary Diagnosis not being showed in the Encounter details

### PR DESCRIPTION
**Pull request by HerbertYiga**
Primary Diagnosis not being showed in the Encounter details 

Ticket ID
https://issues.openmrs.org/browse/RA-1714

### CODE REVIEW

**Comment by dkayiwa** [comment]
Is this still a work in progress pr?

**Comment by HerbertYiga** [comment]
Hi @ibacher  @dkayiwa ,am trying to access the last saved diagnosis after filling in a visit note for a patient, am trying to see whether i can get to a method that at least looks like encounter.getObsAtTopLevel(false) but for this case round,i want to access the diagnosis on the encounter but not on the obs,reading around,i have seen suggestions that it does exist for openmrs-core 2.2,unfortuantely even on upgrading my core apps to platform 2.2 i cant seem to get such a method,do we have diagnosis existing on an encounter currently? thanks

**Comment by ibacher** [comment]
@HerbertYiga I'm not *entirely* sure I understand the context of what you're trying to do, but surely `org.openmrs.api.DiagnosisService#getPrimaryDiagnosis(encounter)` is, ultimately, what you want to populate this with? Prior to 2.2, this would by from the `DiagnosisService` class in the EMR API module.

File: omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java [hunk 1]
```diff
@@ -171,14 +172,15 @@ private SimpleObject parseDisposition(DispositionDescriptor dispositionDescripto
 		return simpleObject;
 	}
 	
-	private SimpleObject parseDiagnosis(DiagnosisMetadata diagnosisMetadata, Obs obs) {
-		Diagnosis diagnosis = diagnosisMetadata.toDiagnosis(obs);
+	private SimpleObject parseDiagnosis(Encounter encounter) {
+		
+		Diagnosis diagnosis =Context.getDiagnosisService().getPrimaryDiagnosis(encounter);
```
Comment (HerbertYiga) [inline omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java:177]: @ibacher   hi Ian, so i want to make sure that the returned diagnosis doesn't depend on the obs but rather the encounter,as shown in this line

File: web-1.9/pom.xml [hunk 2]
```diff
@@ -13,7 +13,7 @@
     <description>1.9 web project for coreapps</description>
 
     <properties>
-        <openMRSVersion>1.9.9</openMRSVersion>
+        <openMRSVersion>2.2.0</openMRSVersion>
```
Comment (HerbertYiga) [inline web-1.9/pom.xml:16]: @ibacher  in order to intoduce (getPrimaryDiagnosis(encounter)), changed the core version of openmrs, though still it shows that it does not exist

File: omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java [hunk 3]
```diff
@@ -45,6 +49,9 @@
     private LocationService locationService;
 
     private DispositionService dispositionService;
+    
+    @Autowired
+    DiagnosisServiceCompatibility diagnosisServiceCompatibility;
```
Comment (HerbertYiga) [inline omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/ParserEncounterIntoSimpleObjects.java:54]: hi @dkayiwa,as i want to use getPrimaryDiagnoses(encounter), i followed the link you dropped me and accomplished the following steps.
   1) created an interface called DiagnosisServiceCompatibility

  2)I implemented this in a submodule,code can be seen in DiagnosisServiceCompatibility2_2 class

3) I then tried to autowire the interface as see here,but i run into  https://pastebin.com/Faj3eLCL on building locally.

Could i be missing out some thing in my poms?

**Comment by sherrif10** [comment]
cc @HerbertYiga  did you travis failure

**Pull request closed without merging**